src/Tasks.UnitTests/ResourceHandling/GenerateResource_Tests.cs

```diff
@@ -3395,8 +3396,36 @@ public static void AssertLogContainsResource(GenerateResource t, string messageI
         {
             Assert.Contains(
                 String.Format(AssemblyResources.GetString(messageID), replacements),
-                ((MockEngine)t.BuildEngine).Log
-                );
+                ((MockEngine) t.BuildEngine).Log
+            );
+        }
+
+        /// <summary>
+        /// Looks for a formatted message in the output log for the task execution, with unknown formatted parameters.
+        /// If verifies that all constant segments of unformatted message are present.
+        /// </summary>
+        public static void AssertLogContainsResourceWithUnspecifiedReplacements(GenerateResource t, string messageID)
+        {
+            var unformattedMessage = AssemblyResources.GetString(messageID);
+            var matches = Regex.Matches(unformattedMessage, @"\{\d+.*?\}");
```

Comment:
Wondering if, instead of matching the parts between placeholders, you could transform the unformatted message to one regex, for example:

```Error, {0} is broken.``` -> ```Error, .* is broken.``` and then do try to match the log against it.

Reply:
Reworked

Reply:
I would probably still go with a single regex as it's so much simpler.

```C#
string messageRegex = Regex.Replace(Regex.Escape(unformattedMessage), "\\\\\\{\\d+\\}", ".*");
```

and then just match log against this.

Working code in https://dotnetfiddle.net/fN7HNz